### PR TITLE
Generic parallel linear solves

### PR DESCRIPTION
**Pull request by renemilk**
Current state is working, but there's a few TODOs left:
 - Hardcoded to four DOFs and square elements
 - no tests are executed in parallel yet

### CODE REVIEW

**Comment by renemilk** [comment]
Most of the tests are now run in parallel. Hardcode is gone too

File: dune/gdt/spaces/localview.hh [hunk 1]
```diff
@@ -0,0 +1,106 @@
+//
+// Created by r_milk01 on 29.03.17.
+//
```
Comment (ftalbrecht) [inline dune/gdt/spaces/localview.hh]: missing header

File: dune/gdt/spaces/rt/dune-pdelab-wrapper.hh [hunk 2]
```diff
@@ -66,6 +66,7 @@ public:
   static_assert(polOrder == 0, "Untested!");
   static_assert(rangeDim == GridLayerType::dimension, "Untested!");
   static_assert(rangeDimCols == 1, "Untested!");
+  static const constexpr Backends backend_type{Backends::pdelab};
```
Comment (ftalbrecht) [inline dune/gdt/spaces/rt/dune-pdelab-wrapper.hh]: s.a.

File: examples/parallel_discretizations/block_swipdg.cc [hunk 3]
```diff
@@ -0,0 +1,322 @@
+#define ALUGRIDDEBUG 1
```
Comment (ftalbrecht) [inline examples/parallel_discretizations/block_swipdg.cc]: missing header

File: dune/gdt/spaces/rt/dune-pdelab-wrapper.hh (hunk 2 above)
Comment (ftalbrecht) [reply dune/gdt/spaces/rt/dune-pdelab-wrapper.hh]: What I meant (and wrote at another place above, at least I thought so) was: I would prefer sticking to
```c++
static const constexpr Backends backend_type = Backends::pdelab;
```
which I find more readable somehow, if you have no objections

File: cluster/phase2.pure.tpl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  procs = 28 * nodes
+  #print('nodes {} - procs {}'.format(nodes, procs))
+  fn = 'node_{}.submit'.format(nodes)
+  open(fn, 'wt').write(tpl.render(**locals()))
+  print('llsubmit {}'.format(fn))
+
+
+for i in range(1,3):
+  output_run(i)
+
```
[the remark this answers is not in the document]
Comment (renemilk) [reply cluster/phase2.pure.tpl]: Indeed.

File: dune/gdt/spaces/rt/dune-pdelab-wrapper.hh (hunk 2 above)
Comment (renemilk) [reply dune/gdt/spaces/rt/dune-pdelab-wrapper.hh]: I don't mind if you change that personally, but it's actually good practice to use it.

File: .gitignore [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 # License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
 #      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
 #          with "runtime exception" (http://www.dune-project.org/license.html)
 # Authors:
 #   Felix Schindler (2011 - 2014, 2016 - 2017)
 #   Rene Milk       (2017)
+#
+# Copyright 2010-2017 dune-gdt developers and contributors. All rights reserved.
+#      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
+#          with "runtime exception" (http://www.dune-project.org/license.html)
```
[the remark this answers is not in the document]
Comment (renemilk) [reply .gitignore]: Fixed now

File: dune/gdt/spaces/localview.hh [hunk 6]
```diff
@@ -0,0 +1,113 @@
+// This file is part of the dune-gdt project:
+//   https://github.com/dune-community/dune-gdt
+// Copyright 2010-2018 dune-gdt developers and contributors. All rights reserved.
+// License: Dual licensed as BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+//      or  GPL-2.0+ (http://opensource.org/licenses/gpl-license)
+//          with "runtime exception" (http://www.dune-project.org/license.html)
+// Authors:
+//   Rene Milk (2017 - 2018)
+//
+// Created by r_milk01 on 29.03.17.
```
Comment (ftalbrecht) [inline dune/gdt/spaces/localview.hh]: drop these two lines?

File: dune/gdt/spaces/rt/interface.hh [hunk 7]
```diff
@@ -179,6 +179,11 @@ public:
   {
     DUNE_THROW(NotImplemented, "RT spaces do not implement constraints!");
   }
+
+  static constexpr bool associates_data_with(int codim)
+  {
+    return codim < dimDomain;
```
Comment (ftalbrecht) [inline dune/gdt/spaces/rt/interface.hh]: This is actually (see Brezzi, Fortin, 1991, p. 117):
* order 0: codim 1
* orders >= 1: codim 0 and 1

**Pull request merged**